detection_rules/rule.py

```diff
@@ -568,7 +568,7 @@ def get_beats_schema(self, index: list, beats_version: str, ecs_version: str) ->
     def get_endgame_schema(self, index: list, endgame_version: str) -> Optional[endgame.EndgameSchema]:
         """Get an assembled flat endgame schema."""
 
-        if "endgame-*" not in index:
+        if index and "endgame-*" not in index:
```

Comment:
really this should be handled from the caller (`index or []`) since we type hint this as a required list. But not the worst to handle it at both ends to be safe

Reply:
Updated on the caller.